ui/src/components/ConfigurationForm/ConfigurationForm.consts.ts

```diff
@@ -124,28 +128,28 @@ export const createStarterValidationSchema = (isScalaVersionFieldVisible: boolea
       // NOTE: unfortunately this is the only way of multiple .when cases in yup :shrug-emoji:
       implementation: yup
         .mixed()
-        .when('effect', {
+        .when(['effect'], {
           is: EffectType.Future,
           then: schema =>
             schema.oneOf(
               mapEffectTypeToEffectImplementation(EffectType.Future),
-              getEffectImplementationFieldMessage(EffectType.Future)
+              getEffectImplementationFieldMessage(EffectType.Future, ScalaVersion.Scala2)
```

Comment:
@lubarskyy  - I am not sure if this is properly handled. Can You look on UI changes?